pkg/controller/node/cidr_set_test.go

```diff
@@ -26,124 +26,167 @@ import (
 )
 
 func TestCIDRSetFullyAllocated(t *testing.T) {
-	_, clusterCIDR, _ := net.ParseCIDR("127.123.234.0/30")
-	a := newCIDRSet(clusterCIDR, 30)
-
-	p, err := a.allocateNext()
-	if err != nil {
-		t.Fatalf("unexpected error: %v", err)
-	}
-	if p.String() != "127.123.234.0/30" {
-		t.Fatalf("unexpected allocated cidr: %s", p.String())
-	}
-
-	_, err = a.allocateNext()
-	if err == nil {
-		t.Fatalf("expected error because of fully-allocated range")
-	}
-
-	a.release(p)
-	p, err = a.allocateNext()
-	if err != nil {
-		t.Fatalf("unexpected error: %v", err)
-	}
-	if p.String() != "127.123.234.0/30" {
-		t.Fatalf("unexpected allocated cidr: %s", p.String())
-	}
-	_, err = a.allocateNext()
-	if err == nil {
-		t.Fatalf("expected error because of fully-allocated range")
+	cases := []struct {
+		clusterCIDRStr string
+		subNetMaskSize int
+		expectedCIDR   string
+	}{
+		{
+			clusterCIDRStr: "127.123.234.0/30",
+			subNetMaskSize: 30,
+			expectedCIDR:   "127.123.234.0/30",
+		},
+		{
+			clusterCIDRStr: "beef:1234::/30",
+			subNetMaskSize: 30,
+			expectedCIDR:   "beef:1234::/30",
+		},
 	}
-}
+	for _, tc := range cases {
+		_, clusterCIDR, _ := net.ParseCIDR(tc.clusterCIDRStr)
+		a := newCIDRSet(clusterCIDR, tc.subNetMaskSize)
 
-func TestCIDRSet_RandomishAllocation(t *testing.T) {
-	_, clusterCIDR, _ := net.ParseCIDR("127.123.234.0/16")
-	a := newCIDRSet(clusterCIDR, 24)
-	// allocate all the CIDRs
-	var err error
-	cidrs := make([]*net.IPNet, 256)
-
-	for i := 0; i < 256; i++ {
-		cidrs[i], err = a.allocateNext()
+		p, err := a.allocateNext()
 		if err != nil {
 			t.Fatalf("unexpected error: %v", err)
 		}
-	}
+		if p.String() != tc.expectedCIDR {
+			t.Fatalf("unexpected allocated cidr: %s", p.String())
+		}
 
-	_, err = a.allocateNext()
-	if err == nil {
-		t.Fatalf("expected error because of fully-allocated range")
-	}
-	// release them all
-	for i := 0; i < 256; i++ {
-		a.release(cidrs[i])
-	}
+		_, err = a.allocateNext()
+		if err == nil {
+			t.Fatalf("expected error because of fully-allocated range")
+		}
+
+		a.release(p)
 
-	// allocate the CIDRs again
-	rcidrs := make([]*net.IPNet, 256)
-	for i := 0; i < 256; i++ {
-		rcidrs[i], err = a.allocateNext()
+		p, err = a.allocateNext()
```

Comment:
So if it allocates a bunch, but then fails, before it has done 256, will we leak the ones that were successful?